sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -278,6 +278,8 @@ private[hive] class HiveClientImpl(
     state.getConf.setClassLoader(clientLoader.classLoader)
     // Set the thread local metastore client to the client associated with this HiveClientImpl.
     Hive.set(client)
+    // Replace conf in the thread local Hive with current conf
+    Hive.get(conf)
```

Comment:
Because of reusing`IsolatedClientLoader.cachedHive`. Without this line, it may use an out-of-date HiveConf. The failed test is `HiveSparkSubmitSuite.test("SPARK-18360: default table path of tables in default database should depend on the " +
    "location of default database")`